src/core.jl

```diff
@@ -134,6 +134,54 @@ Note: An exception will be raised if the edge is not in the graph.
 """
 rem_edge!(g::SimpleGraph, src::Int, dst::Int) = rem_edge!(g, Edge(src,dst))
 
+"""Remove the vertex `v` from graph `g`.
+This operation has to be performed carefully if one keeps external data structures indexed by
+edges or vertices in the graph, since internally the removal is performed swapping the vertices `v`  and `n=nv(g)`,
+and removing the vertex `n` from the graph.
+After removal the vertices in the ` g` will be indexed by 1:n-1.
+This is an O(k) operation, where `k` is the max of the degree of vertices `v` and `n`.
+Note: An exception will be raised if the vertex `v`  is not in the `g`.
```

Comment:
This is a bit misleading, since `rem_edge!()` is not O(1), and you're calling it `k` + `n` times.